instrumentation/spring/starters/spring-boot-starter/build.gradle.kts

```diff
@@ -1,15 +1,20 @@
 plugins {
+  id("io.spring.dependency-management") version "1.1.6"
   id("otel.java-conventions")
   id("otel.publish-conventions")
 }
 
 group = "io.opentelemetry.instrumentation"
 
-val springBootVersion = "2.6.15"
+dependencyManagement {
+  imports {
+    mavenBom("org.springframework.boot:spring-boot-dependencies:2.6.15")
+  }
+}
 
 dependencies {
-  compileOnly("org.springframework.boot:spring-boot-starter:$springBootVersion")
-  compileOnly("org.springframework.boot:spring-boot-starter-aop:$springBootVersion")
+  api("org.springframework.boot:spring-boot-starter")
+  api("org.springframework.boot:spring-boot-starter-aop")
```

Comment:
From the the [Gradle documentation](https://docs.gradle.org/current/userguide/java_library_plugin.html) about `api`: "This is where you declare dependencies which are transitively exported to consumers, for compile time and runtime." 

Let's take the example of a Spring Boot 3.x user whose project does not declare the Spring Boot BOM, or having a Maven file without `spring-boot-starter-parent` as parent. With `api` the dependeny is transitevely exported, so the project would have Spring Boot 3.x dependencies except for `spring-boot-starter-aop`. So,  unexpected results could happen because of a a misalignment of the Spring Boot versions.

So, declaring a Spring Boot dependency as `api` in the OTel Java instrumentation project would force the users to declare the Spring Boot BOM or use `spring-boot-starter-parent` as parent in Maven, to avoid unexpected behaviors.

Reply:
I think we can use `implementation` too - would that address the issues?